azext_iot/central/providers/device_provider.py

```diff
@@ -191,11 +191,14 @@ def get_device_registration_info(
             dps_state = dps_global_service.get_registration_state(
                 id_scope=id_scope, key=key, device_id=device_id
             )
-        dps_state = self.dps_populate_essential_info(dps_state, device.device_status)
+        dps_state = self.dps_populate_essential_info(
```

Comment:
I still think that `self.dps_populate_essential_info` should take the `device_status` as an enum rather than as a string so this function reduces to `self.dps_populate_essential_info(dps_state, device.device_status`, no need to get the `.value` out of it